packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -1019,20 +1032,19 @@ mixin SchedulerBinding on BindingBase {
       return;
     }
 
+    ensureFrameCallbacksRegistered();
```

Comment:
Should this be delayed until right before we call `forceSyncFrame`? Since `forceSyncFrame` is called in a `Timer.run` in theory a real frame could now slip in after these have been registered and before `forceSyncFrame` is called.